main/include/mpu.h

```diff
@@ -2,16 +2,17 @@
 
 namespace mpu {
 
+template<typename T>
 struct Position {
-    float x{};
-    float y{};
-    float z{};
+    T x{};
+    T y{};
+    T z{};
 };
 
 struct Data {
-    Position max{};
-    Position avg{};
-    Position rot{};
+    Position<float> max{};
+    Position<float> avg{};
+    Position<float> rot{};
```

Comment:
Don't you generic code, we don't need it as we are creating software for a very specific hardware configuration

Reply:
Generic code was added apropos https://github.com/grupacosmo/cosmorocket/pull/35#discussion_r1827084786 following DRY principle. If I understand correctly, I was supposed to create separate class for int16_t and float?

Reply:
You can do that or, since those values are temporary, you can just create 2 arrays,  you can also use built-in types that MPU6050 delivers like quaterions or VectorInt16 and VectorFloat if you really don't want to declare another structure just to use it once, there are many was to do it properly without complicating the code. And DRY doesn't mean you have to make every single piece of code generic, it's more of a loose guideline

Reply:
Replaced with library solution in [6be3dc6](https://github.com/grupacosmo/cosmorocket/pull/35/commits/6be3dc6fc090eb45284ac26636f2af69065906d2).
Why exactly arrays are preferred instead of separate variables? In my opinion even the shortest variable name (if named properly) conveys more information than array name with mystical index.

Reply:
> Replaced with library solution in [6be3dc6](https://github.com/grupacosmo/cosmorocket/pull/35/commits/6be3dc6fc090eb45284ac26636f2af69065906d2). Why exactly arrays are preferred instead of separate variables? In my opinion even the shortest variable name (if named properly) conveys more information than array name with mystical index.

Because std::array is guaranteed to be zero-initialized, separate variables are not - this is very bug prone and should be avoided and variables in this context are related to each other and code should show this relation not only formally (by naming them in a specific way) but also programmatically (by wrapping them together)